tlslite/tlsrecordlayer.py

```diff
@@ -722,6 +725,31 @@ def _getMsg(self, expectedType, secondaryType=None, constructorType=None):
                                 yield result
                             continue
 
+                    # If we received a heartbeat request ...
+                    if recordHeader.type == ContentType.heartbeat:
+                        # If heartbeat extension was not negotiated, then
+                        # we send unexpected_message fatal alert
+                        if not self.heartbeat:
+                            for result in self._sendError( \
+                                    AlertDescription.unexpected_message,
```

Comment:
if heartbeats are not enabled, won't this be handled by the default case below (the one with `#Otherwise`) if we enter the ContentType.hearbeat case only if heartbeats were negotiatied?